### DISCUSSION TITLE
Any way to customize shape for each column in stacked bar chart

### DISCUSSION THREAD

**Discussion by siyingch86** [patrykandpatrick/vico#636] [Questions] [answered]
Hi there, I have a question over stacked bar shape. Currently, I've set them to pill shape and get a chart like Pic1. 
<img width="157" alt="Screenshot 2024-03-27 at 3 36 54 PM" src="https://github.com/patrykandpatrick/vico/assets/160801134/0162f6df-1b00-4319-a3fe-c515b9b35cb1">

However, my ideal chart is like Pic2, where the bar attached with each other. And another requirement I want to achieve is for a bar with single group, it will automatically render the pill shape. 
<img width="154" alt="Screenshot 2024-03-27 at 3 37 12 PM" src="https://github.com/patrykandpatrick/vico/assets/160801134/1fbdb64b-8e39-4c87-8d03-8a229f0295ca">

I noticed how show case charts do is the following that we need to adjust the shape according to the index. However, this won't meet another requirement when there's a bar with a single group(another group value is 0), it will show the fixed shape instead of automatically rendering it to pill shape(Pic3 ).          
<img width="162" alt="Screenshot 2024-03-27 at 3 37 58 PM" src="https://github.com/patrykandpatrick/vico/assets/160801134/571c0c65-570f-48dd-bf47-aee515321c02">
  
          
```
rememberColumnCartesianLayer(
                        columns =
                        remember(defaultColumns) {
                            defaultColumns.mapIndexed { index, defaultColumn ->
                                Log.v("TAG", "index is ${index}")
                                val topCornerRadiusPercent =
                                    if (index == defaultColumns.lastIndex) {
                                        Defaults.COLUMN_ROUNDNESS_PERCENT
                                    } else {
                                        0
                                    }
                                val bottomCornerRadiusPercent =
                                    if (index == 0) Defaults.COLUMN_ROUNDNESS_PERCENT else 0
                                LineComponent(
                                    defaultColumn.color,
                                    defaultColumn.thicknessDp,
                                    Shapes.roundedCornerShape(
                                        topCornerRadiusPercent,
                                        topCornerRadiusPercent,
                                        bottomCornerRadiusPercent,
                                        bottomCornerRadiusPercent,
                                    ),
                                )
                            }
                        },
                        mergeMode = { ColumnCartesianLayer.MergeMode.Stacked },
```

**Comment by patrickmichalik** [comment, marked as answer]
Hello! [Vico 2.0.0 Alpha 13](https://github.com/patrykandpatrick/vico/releases/tag/v2.0.0-alpha.13) makes this possible. With `ColumnCartesianLayer.ColumnProvider`, you can customize columns individually. I’d suggest creating a map that stores the column count for each _x_ value and saving it as an extra. You can then retrieve the map in `ColumnProvider#getColumn` and use it to return the appropriate `LineComponent` for each column, taking into account whether there’s another column in its stack. Let me know if you have any questions.
